packages/helpers/src/validate.ts

```diff
@@ -0,0 +1,94 @@
+import { BytesLike, number } from "@ckb-lumos/codec";
+import { createTransactionFromSkeleton, TransactionSkeletonType } from ".";
+import { bytify, hexify } from "@ckb-lumos/codec/lib/bytes";
+import blake2b from "blake2b";
+import { blockchain } from "@ckb-lumos/base";
+
+const { RawTransaction } = blockchain;
+
+type HashAlgorithm = "ckb-blake2b-256" | "ckb-blake2b-160";
+
+interface Hasher {
+  update(data: BytesLike): Hasher;
+  digest(): string;
+}
+
+class Blake2bHasher implements Hasher {
+  constructor(
+    outlength: number,
+    key: Uint8Array | undefined,
+    salt: Uint8Array | undefined,
+    personal: Uint8Array | undefined
+  ) {
+    this.hasher = blake2b(outlength, key, salt, personal);
+  }
+  private hasher: blake2b.Blake2b;
+
+  update(data: BytesLike) {
+    this.hasher.update(bytify(data));
+    return this;
+  }
+
+  digest() {
+    return hexify(this.hasher.digest());
+  }
+}
+
+function createHasher(algorithm: HashAlgorithm): Hasher {
+  switch (algorithm) {
+    case "ckb-blake2b-256":
+      return new Blake2bHasher(
+        32,
+        undefined,
+        undefined,
+        Buffer.from("ckb-default-hash")
+      );
+
+    case "ckb-blake2b-160":
+      return new Blake2bHasher(
+        20,
+        undefined,
+        undefined,
+        Buffer.from("ckb-default-hash")
+      );
+    default:
+      throw new Error(`Unsupported hash algorithm: ${algorithm}`);
+  }
+}
+
+function getWitnessLength(witnesses: Uint8Array) {
+  return number.Uint64.pack(witnesses.length).buffer;
+}
+
+/**
+ * Validate a P2PKH(Pay to public key hash) message
+ * @param messagesForSigning the message digest for signing. means hash(rawTransaction | extraData).
+ * @param rawTransaction raw transaction object
+ * @param hashContentExceptRawTx content to be hashed other than rawTransaction, is generally processed witness
+ * @param hashAlgorithm hash algorithm for signing. Default is `"ckb-blake2b-256"`
+ * @returns the validate result. unless all messages equals, it will return false.
+ */
+export function validateP2PKHMessage(
+  messagesForSigning: BytesLike,
+  rawTransaction: TransactionSkeletonType,
+  hashContentExceptRawTx: BytesLike,
```

Comment:
Is it possible to merge `hashContentExceptRawTx` and `rawTransaction` into one parameter, cause everything it needs to calculate message is in transactionSkeleton.

BTW, this implementation seems unable to handle a case that the tx has multiple witness like: `[witness1, witness2]`

Reply:
> 
BTW, this implementation seems unable to handle a case that the tx has multiple witness like: [witness1, witness2]
In current version. support it is not in current stage. because the hash algorithm may be multiple in this stage(We want this function not only support CKB, but also support other transaction

Reply:
> BTW, this implementation seems unable to handle a case that the tx has multiple witness like: [witness1, witness2]
One signing entries just have one message and `hashContentExceptRawTx`